src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -23,11 +31,17 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
     optionsDisplayProperty,
     hasOther,
     default: defaultOption,
-  } = parameters
+  } = parameters as {
+    label?: string
+    description?: string
+    placeholder?: string
+    search?: boolean
+    hasOther?: boolean
+    default: any
+  } & (ObjectOptions | StringOptions)
 
   const [selectedIndex, setSelectedIndex] = useState<number>()
-  //saving selected value in local state, so that when options are re rendered, we retain this value
-  const [addedOptions, setAddedOptions] = useState<Array<{}>>([])
+  const [addedOption, setAddedOption] = useState<string | null>(null)
```

Comment:
Ended up being easier to keep the custom option as a single value rather than an array. If a new custom value is added, this just gets replaced.